dvc/command/stage.py

```diff
@@ -255,7 +255,8 @@ def _add_common_args(parser):
         "--checkpoints",
         action="append",
         default=[],
-        help=argparse.SUPPRESS,
+        help="Declare output file or directory that will be be handled as an "
+        "intermediate checkpoint upon exp run.",
```

Comment:
Any reason to leave out "Note that checkpoints are not supported by `dvc repro`."?

Reply:
Does `repro` error-out or just ignore them?

Reply:
Errors out:

```
ERROR: failed to reproduce 'dvc.yaml': Checkpoint stages are not supported in 'dvc repro'. Checkpoint stages must be reproduced with 'dvc exp run' or 'dvc exp resume'.
```

Reply:
OK my suggested text: ``"Declare checkpoint output (file or directory) for `dvc exp run`. Not compatible with `dvc repro`"``